spec/git_spec.rb

```diff
@@ -222,6 +222,30 @@ def ensure_only_one_ref(folder)
             should.not.raise { downloader.download }
           end
         end
+
+        describe 'get remote default branch if not set branch with shallow clones' do
+          it 'with git <= 2.10.x' do
```

Comment:
where is this version enforced within the tests? I do not see it.

Reply:
I only try to follow the same way of this  unit-test 
https://github.com/CocoaPods/cocoapods-downloader/blob/bce1d26b422e0efa967f66df955a05a8bbb49a3f/spec/git_spec.rb#L202

Reply:
I only try to follow the same way of this  unit-test 
https://github.com/CocoaPods/cocoapods-downloader/blob/bce1d26b422e0efa967f66df955a05a8bbb49a3f/spec/git_spec.rb#L202

Reply:
oh interesting, had no idea....curious how this happens.....

Reply:
@dimohamdy I checked out your branch and it seems even without your change the tests here are still passing? Can you confirm we are testing the right thing and these tests fail without your change?

Reply:
Hi @dnkoutso, Git shallow is working fine when branch is set. The real issue is caused by `self.preprocess_option(options)` method. This method is deleting the `:branch` option from `options`, and add `:commit` option into `options` and after that it causing to pass adding `--depth 1` parameter when git clone `shallow_clone && !options[:commit]`. We should add test for `preprocess_options`